xbmc/filesystem/File.h

```diff
@@ -72,6 +72,16 @@ class CFile
    *         or undetectable error occur, -1 in case of any explicit error
    */
   ssize_t Read(void* bufPtr, size_t bufSize);
+  /**
+   * Attempt to read a line up to iLineLength bytes from currently opened file into
+   * szLine. If a newline isread, it is stored into the buffer. A terminating null
+   * byte is stored after the last character in the buffer, therefore szLine has to
+   * be at least of length iLineLength + 1.
+   * @param iLineLength pointer to string buffer
+   * @param szLine      size of the buffer
+   * @return True if any bytes were read and stored in szLine, False if no bytes
+   *         are available to read (end of file was reached) or error occur
+   */
```

Comment:
Good documentation, this makes it clear what should happen. I assume this is the same behavior that the standard library uses? Would that be worth referencing?

Also "isread" -> "is read"

Reply:
I don't really understand the change here. It makes it less obvious to the user what happens. The original method would fail if the `iLineLength==0` making it so that there is no out of bounds write for the null terminator.

The method implemented in this PR makes it ok to write out of bounds of the size of the buffer (or in the case you lie about the size and say it's smaller than it is). I don't think that is a good implementation.

I think we should either change the loop `while (iLineLength > 1)` to always save room for a null terminator (and possibly truncate the read) or always just write the null terminator within the bounds `szLine[iLineLength - 1] = 0` (and truncate the read). Or we make sure that the buffer is large enough for the entire read and the null terminator and fail otherwise (like what happens originally).

I'm also not sure what the point of including the newline character in the returned string is?

Reply:
That's the implementation of ReadString() of [IFile](https://github.com/xbmc/xbmc/blob/26adddf24d919b812a1bedf2507d5d33d4ebdbcf/xbmc/filesystem/IFile.cpp) and [CCurFile](https://github.com/xbmc/xbmc/blob/26adddf24d919b812a1bedf2507d5d33d4ebdbcf/xbmc/filesystem/CurlFile.cpp#L1251). It is similar to fgets().

Only the implementation of chunked reads over m_pBuffer differs.

Reply:
@garbear Typo is fixed, thanks

Reply:
I agree that requiring the buffer to be larger then  size parameter is not intuitive. Also the documentation is wrong, the parameter names are swapped. Plus it says <code>@param ~~szLine~~iLineLength      size of the buffer</code> when it says above that it has to be ` size of the buffer - 1` or less.

> I think we should either change the loop while (iLineLength > 1) to always save room for a null terminator (and possibly truncate the read) or always just write the null terminator within the bounds szLine[iLineLength - 1] = 0 (and truncate the read).

I agree with this, the first method being preferred IMO.

> Or we make sure that the buffer is large enough for the entire read and the null terminator and fail otherwise (like what happens originally).

I think that's the worse choice. If we would require the caller to pass a large enough buffer then the method would need to communicate the required size somehow in case of an error. Otherwise the caller has to guess. Also the new behaviour is consistent with the other `ReadString(...)` methods.

Reply:
@neo1973 Thanks. Parameters are corrected.

Reply:
> > Or we make sure that the buffer is large enough for the entire read and the null terminator and fail otherwise (like what happens originally).
> 
> 
> 
> I think that's the worse choice. If we would require the caller to pass a large enough buffer then the method would need to communicate the required size somehow in case of an error. Otherwise the caller has to guess. Also the new behaviour is consistent with the other `ReadString(...)` methods.

That's not what I meant, I simply mean we fail if the buffer isn't large enough to hold what we want (like in the original implementation).

If the new line character is desired and the point of this PR is to add that then we should do so without compromising the interface.